File: server/feed11/src/main/scala/info/glennengstrand/news/dao/CassandraDao.scala
package info.glennengstrand.news.dao

import com.datastax.oss.driver.api.core.CqlSession
import java.net.InetSocketAddress

object CassandraDao {
  val noSqlHost = sys.env.get("NOSQL_HOST").getOrElse("localhost")
  val noSqlTtl = sys.env.get("NOSQL_TTL").getOrElse(1000 * 60 * 60 * 24)
  val noSqlKeyspace = sys.env.get("NOSQL_KEYSPACE").getOrElse("activity")
  def connect: CqlSession = {
    CqlSession.builder()
    .addContactPoint(new InetSocketAddress(noSqlHost, 9042))
    .withLocalDatacenter("datacenter1")
    .withKeyspace(noSqlKeyspace)
    .build
  }
  lazy val session = connect
}

File: server/feed14/service/DAOs/CassandraDao.cs
using Cassandra;

namespace newsfeed.DAOs;

public class CassandraDao {
    private static Cluster? cluster;
    private static Cassandra.ISession? session;

    private static readonly string keyspace = Environment.GetEnvironmentVariable("NOSQL_KEYSPACE") ?? "activity";

    protected Cassandra.ISession CassandraSession {
        get {
            if (session == null) {
                lock(keyspace) {
                    if (session == null) {
                        cluster = Cluster.Builder().AddContactPoint(Environment.GetEnvironmentVariable("NOSQL_HOST")).Build();
                        session = cluster.Connect(keyspace);
                    }
                }
            }
            return session;
        }
    }
}